src/client.ts

```diff
@@ -268,54 +308,71 @@ export class SubscriptionClient {
     };
 
     this.client.onmessage = ({ data }: {data: any}) => {
-      let parsedMessage: any;
-      try {
-        parsedMessage = JSON.parse(data);
-      } catch (e) {
-        throw new Error(`Message must be JSON-parseable. Got: ${data}`);
-      }
-      const subId = parsedMessage.id;
-      if ([KEEPALIVE, INIT_SUCCESS, INIT_FAIL].indexOf(parsedMessage.type) === -1 && !this.subscriptions[subId]) {
-        this.unsubscribe(subId);
+      this.processReceivedData(data);
+    };
+  }
+
+  private processReceivedData(receivedData: any) {
+    let parsedMessage: any;
+    let reqId: number;
+
+    try {
+      parsedMessage = JSON.parse(receivedData);
+      reqId = parsedMessage.id;
+    } catch (e) {
+      throw new Error(`Message must be JSON-parseable. Got: ${receivedData}`);
+    }
+
+    // Complete every request that are not registered on the client but are being sent by the server and are not
+    // in the allowed list
+    if (
```

Comment:
This is a bit hard to read. I think if the code was easier to understand, the comment wouldn't be needed.